scripts/boxgame.js

```diff
@@ -0,0 +1,119 @@
+class BoxGameComponent {
+    box = {};
+    score = document.createElement("div");
+    squares = [];
+    container = document.createElement('div');
+    gameAreaEl = document.createElement("div");
+    gamebox = {
+        x: Math.floor(800 / 100),
+        y: Math.floor(200 / 100)
+    };
+    player = {
+        speed: 100,
+        square: 1,
+        score: 0
+    };
+
+    init = (button) => {
+        this.gameAreaEl.innerHTML = '';
+        this.container.innerHTML = '';
+        this.score.innerHTML = '';
+        document.body.appendChild(this.container);
+        this.createGameAreaDom();
+        this.addListeners();
+        this.build();
+        button.style.display = 'none';
+    };
+
+    addListeners = () => {
+        document.addEventListener('DOMContentLoaded', this.build);
+        document.addEventListener('keyup', function (e) {
+            const allowKey = {
+                37:'left',
+                38: 'up',
+                39: 'right',
+                40: 'down'
+            };
+            if(allowKey[e.keyCode]) {
+                game.handleKey(allowKey[e.keyCode])
+            }
+        });
+    };
+
+    createGameAreaDom = () => {
+        this.gameAreaEl.setAttribute("class", "gameArea");
+        this.score.setAttribute("class", "score");
+        this.container.appendChild(this.gameAreaEl);
+        this.container.appendChild(this.score);
+        this.gameAreaEl.getBoundingClientRect();
+    };
+
+    setGameArea() {
+        return this.gameAreaEl.getBoundingClientRect();
+    };
+
+    makeActive = () => {
+        let randomIndex = Math.floor(Math.random() * this.squares.length);
+
+        if (randomIndex != 0 && this.player.square != randomIndex){
```

Comment:
you only have to check that the randomIndex is not 0 because this.squares has no element at [0]
better to make sure that it does have a value at [0] as explained in comment below